src/main/java/org/sagebionetworks/bridge/spring/controllers/ParticipantReportController.java

```diff
@@ -244,12 +242,16 @@ public StatusMessage saveParticipantReportForWorker(@PathVariable String identif
             throw new BadRequestException("A health code is required to save report data.");
         }
         String healthCode = node.get("healthCode").asText();
-        
+
+        // NOTE: Use of session.getAppId() here and below doesn't allow the worker to switch accounts.
+        // This is an error in the API.
+        String userId = accountService.getAccountId(session.getAppId(), "healthCode:"+healthCode)
+                .orElseThrow(() -> new EntityNotFoundException(Account.class));
```

Comment:
File a JIRA on this.

Reply:
No, it has to work this way because all the worker had for this call was a health code.